controller/JobRoleController.ts

```diff
@@ -29,4 +32,34 @@ module.exports = function(app: Application) {
             }
             res.render('view-jobrole-specification', { jobRoleSpec: data } )
     })
-}
\ No newline at end of file
+    app.get('/add-jobrole', async (req: Request, res: Response) => {
+        let bands = [];
+        let capabilitys = [];
+
+        try {
+            bands = await BandService.getBands() 
+            capabilitys = await CapabilityService.getCapabilities()
+        } catch (e) {
+            console.error(e);
+        }
+        res.render('add-jobrole', { bands: bands, capabilitys: capabilitys  })
+    })
+
+    app.post('/add-jobrole', async (req: Request, res: Response) => {
+        let data: JobRoleCorrect = req.body
+        
+        let id: Number
+
+        try {
+            id = await jobRoleService.createJobRole(data)
+
+            res.redirect('/job_roles')
+        } catch (e) {
+            console.error(e);
+
+            res.locals.errormessage = e.message
+
+            res.render('add-jobrole')
```

Comment:
when an error is thrown you're not pre-populating the dropdown lists again or passing through the data that had been submitted on the form

Reply:
comment still needs addressed